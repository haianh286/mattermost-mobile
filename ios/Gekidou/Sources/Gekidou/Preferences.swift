//
//  File.swift
//  
//
//  Created by Elias Nahum on 26-06-22.
//

import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
