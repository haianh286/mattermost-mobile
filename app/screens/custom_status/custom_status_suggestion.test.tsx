// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {TouchableOpacity} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';
import CustomStatusSuggestion from '@screens/custom_status/custom_status_suggestion';

describe('screens/custom_status_suggestion', () => {
    const baseProps = {
        handleSuggestionClick: jest.fn(),
        emoji: 'calendar',
        text: 'In a meeting',
        theme: Preferences.THEMES.default,
        separator: false,
    };

    it('should match snapshot', () => {
        const wrapper = shallow(
            <CustomStatusSuggestion
                {...baseProps}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should match snapshot with separator and clear button', () => {
        const wrapper = shallow(
            <CustomStatusSuggestion
                {...baseProps}
                separator={true}
                handleClear={jest.fn()}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should call handleSuggestionClick on clicking the suggestion', () => {
        const wrapper = shallow(
            <CustomStatusSuggestion
                {...baseProps}
            />,
        );

        wrapper.find(TouchableOpacity).simulate('press');
        expect(baseProps.handleSuggestionClick).toBeCalled();
    });
});
