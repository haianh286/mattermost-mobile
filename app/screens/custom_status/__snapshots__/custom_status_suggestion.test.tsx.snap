// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/custom_status_suggestion should match snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  testID="custom_status_suggestion.In a meeting"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "minHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3d3c40",
          "height": 46,
          "left": 14,
          "marginRight": 6,
          "top": 12,
          "width": 45,
        }
      }
    >
      <Connect(Emoji)
        emojiName="calendar"
        size={20}
      />
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 14,
            "paddingTop": 14,
            "width": "70%",
          }
        }
      >
        <View>
          <CustomStatusText
            text="In a meeting"
            textStyle={
              Object {
                "color": "#3d3c40",
              }
            }
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</ForwardRef>
`;

exports[`screens/custom_status_suggestion should match snapshot with separator and clear button 1`] = `
<ForwardRef
  onPress={[Function]}
  testID="custom_status_suggestion.In a meeting"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "minHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3d3c40",
          "height": 46,
          "left": 14,
          "marginRight": 6,
          "top": 12,
          "width": 45,
        }
      }
    >
      <Connect(Emoji)
        emojiName="calendar"
        size={20}
      />
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 14,
            "paddingTop": 14,
            "width": "70%",
          }
        }
      >
        <View>
          <CustomStatusText
            text="In a meeting"
            textStyle={
              Object {
                "color": "#3d3c40",
              }
            }
            theme={
              Object {
                "awayIndicator": "#ffbc42",
                "buttonBg": "#166de0",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3d3c40",
                "codeTheme": "github",
                "dndIndicator": "#f74343",
                "errorTextColor": "#fd5960",
                "linkColor": "#2389d7",
                "mentionBg": "#ffffff",
                "mentionBj": "#ffffff",
                "mentionColor": "#145dbf",
                "mentionHighlightBg": "#ffe577",
                "mentionHighlightLink": "#166de0",
                "newMessageSeparator": "#ff8800",
                "onlineIndicator": "#06d6a0",
                "sidebarBg": "#145dbf",
                "sidebarHeaderBg": "#1153ab",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#579eff",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#4578bf",
                "sidebarUnreadText": "#ffffff",
                "type": "Mattermost",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.2)",
            "height": 1,
            "marginRight": 16,
          }
        }
      />
    </View>
  </View>
</ForwardRef>
`;
